### PR TITLE
MK3: XYZ cal change message and add LCD ouput

### PR DESCRIPTION
**Pull request by 3d-gussner**
@Hauzman @vintagepc @AttilaSVK @Prime1910 @ingbrzy @shatter136 @Painkiller56 @OS-kar

Please check my changes to the community translations.
:+1:  correct
:-1:  needs change and please comment here

- [x] Create cherry-pick PR to MK3_3.12 branch

### CODE REVIEW

**Review by shatter136 — APPROVED** [review]
looks good to me

**Comment by Painkiller56** [comment]
Changes look good.

**Comment by Painkiller56** [comment]
Changes look good.

 

Från: 3d-gussner ***@***.***> 
Skickat: den 15 december 2022 14:48
Till: prusa3d/Prusa-Firmware ***@***.***>
Kopia: Bernt Ahblbäck ***@***.***>; Mention ***@***.***>
Ämne: [prusa3d/Prusa-Firmware] MK3: XYZ cal change message and add LCD ouput (PR #3830)

 

@Hauzman <https://github.com/Hauzman>  @vintagepc <https://github.com/vintagepc>  @AttilaSVK <https://github.com/AttilaSVK>  @Prime1910 <https://github.com/Prime1910>  @ingbrzy <https://github.com/ingbrzy>  @shatter136 <https://github.com/shatter136>  @Painkiller56 <https://github.com/Painkiller56>  @OS-kar <https://github.com/OS-kar> 

Please check my changes to the community translations.
👍 correct
👎 needs change and please comment here

  _____  


You can view, comment on, or merge this pull request online at:


  https://github.com/prusa3d/Prusa-Firmware/pull/3830


Commit Summary


*	c982f81 <https://github.com/prusa3d/Prusa-Firmware/pull/3830/commits/c982f817a5b678a7883a8b14a539bcbaa1494f4d>  Add LCD output during xyz calibration
*	e077403 <https://github.com/prusa3d/Prusa-Firmware/pull/3830/commits/e077403638eec2c2e6348f7cb5bf093a5870c826>  Change xyz cal message `up to 24 min.` instead `approx. 12 min.`


File Changes 


(21 <https://github.com/prusa3d/Prusa-Firmware/pull/3830/files>  files)

*	M Firmware/ultralcd.cpp <https://github.com/prusa3d/Prusa-Firmware/pull/3830/files#diff-5fbfdca43c192573e31d8e78d255c510fb6e213b118cf38173277ec2a2163880>  (2) 
*	M Firmware/xyzcal.cpp <https://github.com/prusa3d/Prusa-Firmware/pull/3830/files#diff-13adbb7512f6ea84ba8beb7fa67919b470cc43ab68a1d36053497e9ae8750c09>  (12) 
*	M Firmware/xyzcal.h <https://github.com/prusa3d/Prusa-Firmware/pull/3830/files#diff-16cb908f00f2b7b1509c14a7e038e1e662e97b799111e5ee6e959fbb9cea9a77>  (2) 
*	M lang/po/Firmware.pot <https://github.com/prusa3d/Prusa-Firmware/pull/3830/files#diff-14fedb8dfdce511a0bdcf251bbe0fc1c53d69fcec094186519cc6f57826b4b1b>  (2) 
*	M lang/po/Firmware_cs.po <https://github.com/prusa3d/Prusa-Firmware/pull/3830/files#diff-0aa4aafa0e88956cdefd904d3a4d90d8075a6baea4b7a7552fa25663b8550f7c>  (4) 
*	M lang/po/Firmware_da.po <https://github.com/prusa3d/Prusa-Firmware/pull/3830/files#diff-36ea2d7305804fd87c36473416094f537db3a09da66f11d667072b0a5545d8c8>  (2) 
*	M lang/po/Firmware_de.po <https://github.com/prusa3d/Prusa-Firmware/pull/3830/files#diff-7aa3bbee27c68325a9362a2508a9dbdffb40452a4296772d7b6075b7cafd7d43>  (4) 
*	M lang/po/Firmware_es.po <https://github.com/prusa3d/Prusa-Firmware/pull/3830/files#diff-94fdeb509cca7d35e68df03af06931c6c67805acb0c427946922c1b5caa91549>  (4) 
*	M lang/po/Firmware_fr.po <https://github.com/prusa3d/Prusa-Firmware/pull/3830/files#diff-e1c029266b2a50654a6638406d3cfe838eeb4502bde0aadca6cdcb4372d42a44>  (4) 
*	M lang/po/Firmware_hr.po <https://github.com/prusa3d/Prusa-Firmware/pull/3830/files#diff-487d5469566575627c7dc47abf437c52e890de941c1fd8d4db5a2a4d6ef08d64>  (4) 
*	M lang/po/Firmware_hu.po <https://github.com/prusa3d/Prusa-Firmware/pull/3830/files#diff-3a7d51d07fb16c60d273b7986aa88a7892939b19a9bad4f3501a153ffb0604e9>  (4) 
*	M lang/po/Firmware_it.po <https://github.com/prusa3d/Prusa-Firmware/pull/3830/files#diff-d948679300ec3cbaa7e705150dfcaee8932cf6f047e800ca1036ff551f31e81c>  (4) 
*	M lang/po/Firmware_lb.po <https://github.com/prusa3d/Prusa-Firmware/pull/3830/files#diff-f53f25ee2cf26addfe4279bc64925bd50df3d5a6b252fed807ad55efdd4e0753>  (2) 
*	M lang/po/Firmware_lt.po <https://github.com/prusa3d/Prusa-Firmware/pull/3830/files#diff-ad76d3f0e841fe9a6836d117d76225e0342eb1b5c66d934f795659e293ae1826>  (2) 
*	M lang/po/Firmware_nl.po <https://github.com/prusa3d/Prusa-Firmware/pull/3830/files#diff-92de134d7cfe75b62e86a4e83716cd95550b11f2ddee228150a9c6faa4d6a34b>  (4) 
*	M lang/po/Firmware_no.po <https://github.com/prusa3d/Prusa-Firmware/pull/3830/files#diff-22e5dd5d54111e837f3b40db65f2a126b09cc4157dcfa1f34b6e5dde0bf4a607>  (4) 
*	M lang/po/Firmware_pl.po <https://github.com/prusa3d/Prusa-Firmware/pull/3830/files#diff-67132bccb4c23fbeaf41753633298c3047956124c6a42a0a8a574e3c042a09e0>  (4) 
*	M lang/po/Firmware_ro.po <https://github.com/prusa3d/Prusa-Firmware/pull/3830/files#diff-6774f424618c20ccb1d99bf93cb34bab0b6ee03a55868fa7021a11ef6c28667f>  (4) 
*	M lang/po/Firmware_sk.po <https://github.com/prusa3d/Prusa-Firmware/pull/3830/files#diff-9584ba5c6a2b7618c16dc96fa21d692e9d5d69c393c55f37166bb05e98557fbc>  (4) 
*	M lang/po/Firmware_sl.po <https://github.com/prusa3d/Prusa-Firmware/pull/3830/files#diff-f4fd59fd0a45a79de94d003d6fefcc68d821c43b49cca6eefa83a092051913a6>  (2) 
*	M lang/po/Firmware_sv.po <https://github.com/prusa3d/Prusa-Firmware/pull/3830/files#diff-cf46655e0192f28112da2985d28dc728949010730a0ac629dbb7049f9f53a357>  (4) 


Patch Links:


*	https://github.com/prusa3d/Prusa-Firmware/pull/3830.patch
*	https://github.com/prusa3d/Prusa-Firmware/pull/3830.diff

**Comment by Prime1910** [comment]
Looks good👍🏻

**Comment by 3d-gussner** [comment]
You guys are awesome reacting that fast. :hugs:

File: Firmware/xyzcal.cpp [hunk 1]
```diff
@@ -135,9 +135,11 @@ pos_mm_t pos_2_mm(float pos){
 	return pos * 0.01f;
 }
 
-void xyzcal_meassure_enter(void)
+void xyzcal_meassure_center(void)
 {
-	DBG(_n("xyzcal_meassure_enter\n"));
+	DBG(_n("xyzcal_meassure_center\n"));
+	lcd_set_cursor(4,3);
+	lcd_printf_P(PSTR("Measure center  ")); ////MSG_MEASURE_CENTER c=16
```
Comment (leptun) [inline Firmware/xyzcal.cpp]: Why is this one not translated?

File: Firmware/xyzcal.cpp [hunk 2]
```diff
@@ -569,6 +573,8 @@ void __attribute__((noinline)) xyzcal_scan_pixels_32x32_Zhop(int16_t cx, int16_t
 			sm4_set_dir(X_AXIS, d);
             //@size=242
 			DBG(_n("%d\n"), 64 - (r * 2 + d)); ///< to keep OctoPrint connection alive
+			lcd_set_cursor(4,3);
+			lcd_printf_P(PSTR("Countdown: %d "),64 - (r * 2 + d)); ////MSG_COUNTDOWN c=14
```
Comment (leptun) [inline Firmware/xyzcal.cpp]: Why not translated?

File: Firmware/xyzcal.cpp [hunk 3]
```diff
@@ -158,6 +160,8 @@ void xyzcal_meassure_enter(void)
 void xyzcal_meassure_leave(void)
 {
 	DBG(_n("xyzcal_meassure_leave\n"));
+	lcd_set_cursor(4,3);
+	lcd_printf_P(PSTR("                ")); ////MSG_POINT_FINISHED c=16
```
Comment (leptun) [inline Firmware/xyzcal.cpp]: Please use lcd_space(16)

**Review by leptun — COMMENTED** [review]
Please address the questions about the xyzcal.cpp changes. The romanian translation looks correct :+1:

**Comment by Hauzman** [comment]
i have check it to romanian translation looks correct 👍

**Review by OS-kar — APPROVED** [review]
Looks good

File: Firmware/xyzcal.cpp (hunk 1 above)
Comment (3d-gussner) [reply Firmware/xyzcal.cpp]: Try to avoid translations on rare used messages to save space.

File: Firmware/xyzcal.cpp (hunk 2 above)
Comment (3d-gussner) [reply Firmware/xyzcal.cpp]: Try to avoid translations on rare used messages to save space.

File: Firmware/xyzcal.cpp (hunk 1 above)
Comment (leptun) [reply Firmware/xyzcal.cpp]: Please use lcd_puts_at_P if no formatting is needed.

**Comment by AttilaSVK** [comment]
The Hungarian translation is correct

**Review by leptun — APPROVED** [review]

**Pull request merged**